### PR TITLE
Clarify track_running_stats docs; Make SyncBatchNorm track_running_stats behavior consistent

### PR DESCRIPTION
**Pull request by ssnl**
context: https://github.com/pytorch/pytorch/pull/38084

Fixes #{issue number}

### CODE REVIEW

File: torch/nn/modules/batchnorm.py [hunk 1]
```diff
@@ -485,7 +495,25 @@ def forward(self, input: Tensor) -> Tensor:
             else:  # use exponential moving average
                 exponential_average_factor = self.momentum
 
-        need_sync = self.training or not self.track_running_stats
+        r"""
```
Comment (mrshenli) [inline torch/nn/modules/batchnorm.py:498]: Any reason for using rstring instead of comments for the following two doc blocks?

Comment (SsnL) [reply torch/nn/modules/batchnorm.py:498]: hmm not really. They were strings from `_BatchNorm` so I kept them this way when copying.

**Review by mrshenli — APPROVED** [review]
